src/sagemaker/fw_utils.py

```diff
@@ -124,6 +125,43 @@ def validate_source_dir(script, directory):
     return True
 
 
+def validate_source_code_input_against_pipeline_variables(
+    entry_point: Optional[Union[str, PipelineVariable]] = None,
+    source_dir: Optional[Union[str, PipelineVariable]] = None,
+    git_config: Optional[Dict[str, str]] = None,
+):
+    """Validate source code input against pipeline variables
+
+    Args:
+        entry_point (str, PipelineVariable): The path to the local Python source file that
+            should be executed as the entry point to training (default: None).
+        source_dir (str, PipelineVariable): The Path to a directory with any other
+            training source code dependencies aside from the entry point file (default: None).
+        git_config (Dict[str, str]): Git configurations used for cloning files (default: None).
+    """
+    if git_config:
+        if is_pipeline_variable(entry_point) or is_pipeline_variable(source_dir):
+            raise TypeError(
+                "entry_point, source_dir should not be pipeline variables when git_config is given."
+            )
+    if is_pipeline_variable(entry_point):
+        if not source_dir:
+            raise TypeError(
+                "The entry_point should not be a pipeline variable when source_dir is missing."
+            )
+        logger.warning(
+            "The entry_point is a pipeline variable: %s. Its execution-time value has to be "
```

Comment:
its compile time?

Reply:
No, I put execution-time value here on purpose. The thing is eventually for training job to correctly work, the `entry_point` has to be a local path in the container env to a file which is located at the root of `source_dir`.
We're not able to validate this in compile time, so I add a warning to remind users.

Reply:
nit: 

ok, I found this description a bit confusing, how about:

> During pipeline execution, `entry_point` has to be resolved to a value pointing to a local path in the container, blah, blah.

Reply:
Almost missed this comment. Making sense. Updated in the new commit.

Reply:
Updated to:
```
        logger.warning(
            "The entry_point is a pipeline variable: %s. During pipeline execution, "
            "the interpreted value of entry_point has to be a local path in the container, "
            "pointing to a Python source file which is located at the root of source_dir.",
            type(entry_point),
        )
   ```